Add the man pages from the official 0.7.2 release tarball.

Upstream generates these with docbook2man in its make-dist script and
ships them in the release tarball; regenerating them from doc/*.sgml
would require the legacy SGML docbook-utils toolchain as a build
dependency. Ship the release copies so building from the tagged git
source (github.com/ldo/dvdauthor) installs the same documentation the
previous sourceforge release tarball did.
See https://github.com/macaroni-os/mark-issues/issues/764

--- a/doc/dvdauthor.1
+++ b/doc/dvdauthor.1
@@ -0,0 +1,583 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "DVDAUTHOR" "1" "31 December 2016" "" "DVDAuthor Man Pages"
+
+.SH NAME
+dvdauthor \- assembles multiple mpeg program streams into a suitable DVD filesystem
+.SH SYNOPSIS
+
+\fBdvdauthor\fR [ \fB-o \fIoutput-dir\fB\fR ] \fB-x \fIxml-control-file\fB\fR
+
+
+\fBdvdauthor\fR [ \fB-o \fIoutput-dir\fB\fR ] [ \fB-j\fR | \fB--jumppad\fR | \fB-g\fR | \fB--allgprm\fR ] [ \fB-T\fR | \fB--toc\fR ] [ \fB\fImenu or title options\fB\fR ]
+
+.SH "DVD BACKGROUND"
+.PP
+At a high level, a DVD is a collection of menus and titles.
+Conceptually, a menu contains buttons which can be assigned actions
+and provides a list of choices to the end user, while a title contains
+the main content of the DVD.  However, in reality many of the features
+available in menus (including buttons, pausing, and looping) are also
+available in titles.
+.PP
+The menus and titles are divided into titlesets and the VMGM menu set.
+A titleset can contain a number of menus and titles which are meant to
+act together.  The "menu", "audio", "subtitle", and "angle" buttons on
+the DVD player's remote control will all access menus in the same
+titleset as the title which is being played.  All the titles and menus
+of a given titleset have the same video, audio, and subtitle settings
+(the definitions for the menus are independent from the definitions
+for the titles), so if you want to have different settings (for
+example widescreen vs standard aspect ratios), then you need separate
+titlesets.  Titlesets are not meant to jump to one another, so the
+VMGM menu domain is used.  It is a collection of menus (no titles)
+that can access the menus and titles of all the titlesets.
+.PP
+One of the most frustrating things when deciding how to author a DVD
+is that there are often many ways to accomplish the same task.  For
+example, you must decide whether to locate menus at the VMGM level or
+the titleset level.  A typical setup is to locate the high level menus
+at the VMGM level, and the low level configuration menus (scene /
+audio / subtitle selection) at the titleset.  If there are DVD extras,
+perhaps with a lower quality audio track and a 4:3 aspect ratio, then
+they would be in a separate titleset with a menu to select among the
+extras located at the titleset level.
+.SH "DVDAUTHOR DESCRIPTION"
+.PP
+\fBdvdauthor\fR works in discrete operations.  It
+authors each titleset one at a time, and then finally authors the VMGM
+to complete the disc.  At that point the contents can be written out
+to a DVD.  If you are controlling \fBdvdauthor\fR with
+command line arguments, then each step will occur independently;
+however if you are using the XML control file, then you have the
+option of combining some or all the steps into one.
+.PP
+The VOBs passed to \fBdvdauthor\fR must have DVD
+NAV (VOBU) packets multiplexed in at the correct locations.  Many
+tools can do this, including \fBmplex\fR from mjpegtools
+1.6.0 or later.  \fBdvdauthor\fR will then fill these
+packets in with the correct data.  Special care has been taken to
+ensure \fBdvdauthor\fR is fifo compliant; that is every
+source VOB can be the output of another program (such as
+\fBmplex\fR).  This can make execution faster on many
+systems by avoiding extra filesystem accesses.
+.SH "COMMAND LINE DESCRIPTION"
+.TP
+\fB-o \fIoutput-dir\fB\fR
+.TP
+\fB-O \fIoutput-dir\fB\fR
+The destination directory to store the DVD-Video file structure in.
+If -O is specified, then any existing directory is (safely) deleted (only items that
+look like part of a DVD-Video structure are removed).
+.TP
+\fB-x \fIxml-control-file\fB\fR
+Specifies the control file describing the output structure to create.
+.PP
+Note that the remaining command-line options are \fBdeprecated\fR\&. Use the
+XML control file instead.
+.TP
+\fB-j\fR
+.TP
+\fB--jumppad\fR
+Enables the creation of jumppads, which allow greater flexibility in choosing jump/call destinations.
+.TP
+\fB-g\fR
+.TP
+\fB--allgprm\fR
+Enable the use of all 16 general purpose registers.  Prohibits the use of jumppad and some complex expressions that require temporary registers.
+.TP
+\fB-T\fR
+Creates the table of contents file instead of a
+titleset. If this option is used, it should be listed first, and you
+may not specify any titles.
+.TP
+\fB-m\fR
+Creates a menu.
+.TP
+\fB-t\fR
+Creates a title.
+.TP
+\fB-v \fIvideo-opts\fB\fR
+.TP
+\fB--video=\fIvideo-opts\fB\fR
+A plus (+) separated list of video options.  Dvdauthor
+will try to infer any unspecified options.  pal, ntsc, 4:3, 16:9,
+720xfull, 720x576, 720x480, 704xfull, 704x576, 704x480, 352xfull,
+352x576, 352x480, 352xhalf, 352x288, 352x240, nopanscan, noletterbox, crop.
+Default is 4:3, 720xfull
+.TP
+\fB-a \fIaudio-opts\fB\fR
+.TP
+\fB--audio=\fIaudio-opts\fB\fR
+A plus (+) separated list of options for an audio
+track, with each track separated by a comma (,).  For example -a
+ac3+en,mp2+de specifies two audio tracks: the first is an English
+track encoded in AC3, the second is a German track encoded using
+MPEG-1 layer 2 compression.  ac3, mp2, pcm, dts, 16bps, 20bps, 24bps,
+drc, surround, nolang, 1ch, 2ch, 3ch, 4ch, 5ch, 6ch, 7ch, 8ch, and any
+two letter ISO 639 language abbreviation.  Default is 1 track, mp2,
+20bps, nolang, 2ch.  'ac3' implies drc, 6ch.
+.TP
+\fB-s \fIsubpicture-opts\fB\fR
+.TP
+\fB--subpictures=\fIsubpicture-opts\fB\fR
+A plus (+) separated list of options for a subpicture
+track, with each track separated by a comma (,). nolang and any two
+letter language abbreviation (see -a) Default is no subpicture
+tracks.
+.TP
+\fB-e \fIentry(s)\fB\fR
+.TP
+\fB--entry=\fIentry(s)\fB\fR
+Makes the current menu the default for certain
+circumstances. It is a comma separated list of any of:
+
+for
+TOC menus: title
+
+for VTS menus: root, ptt, audio,
+subtitle, angle
+.TP
+\fB-p \fIpalette-file\fB\fR
+.TP
+\fB--palette=\fIpalette-file\fB\fR
+Specifies where to get the subpicture
+palette. Settable per title and per menu. If the filename ends in .rgb
+(case insensitive) then it is assumed to be RGB, otherwise it is
+YUV. Entries should be 6 hexadecimal
+digits. \fIFILE\fR defaults to
+xste-palette.dat
+.TP
+\fB-c \fIchapterpts\fB\fR
+.TP
+\fB--chapters=\fIchapterpts\fB\fR
+Specifies a comma (,) separated list of chapter
+markers. Each marker is of the form [[h:]mm:]ss[.frac] and is relative
+to the SCR of the next file listed (independent of any timestamp
+transposing that occurs within dvdauthor). The chapter markers ONLY
+apply to the next file listed. Defaults to 0.
+.TP
+\fB-f \fImpeg-file\fB\fR
+.TP
+\fB--file=\fImpeg-file\fB\fR
+.TP
+\fB\fImpeg-file\fB\fR
+Specifies either a file, a pipe, or a shell command
+ending in | which supplies an MPEG-2 system stream with VOB sectors
+inserted in the appropriate places (using mplex -f 8 to
+generate)
+.TP
+\fB-b \fIbuttondef\fB\fR
+.TP
+\fB--button=\fIX1xY1-X2xY2,commandlist\fB\fR
+creates a button of the specified size. See
+LANGUAGE DESCRIPTION for a description of \fIcommandlist\fR\&.
+.TP
+\fB-i \fI[pre|post]=commandlist\fB\fR
+.TP
+\fB--instructions=\fI[pre|post]=commandlist\fB\fR
+Executes the \fIcommandlist\fR
+instructions either before or at the end of the menu/title.  See
+LANGUAGE DESCRIPTION for the format of \fIcommandlist.\fR
+.SH "XML DESCRIPTION"
+.PP
+Here is the basic structure of the control file:
+
+.nf
+<dvdauthor [dest="\fIoutput-dir\fR"] [jumppad="1|on|yes" | allgprm="1|on|yes"] [format="ntsc|pal"] [provider="\fIprovider\fR"]>
+   <vmgm>
+      [<fpc>\fIcommands;\fR</fpc>]
+      <menus [lang="\fIlanguage-code\fR"]>
+         <video [format="ntsc|pal"] [aspect="4:3|16:9"]
+                [resolution="\fIXxY\fR"] [caption="field1|field2"]
+                [widescreen="nopanscan|noletterbox|crop"] />
+         <audio [format="mp2|ac3|dts|pcm"] [channels="\fInumchannels\fR"]
+                [quant="16bps|20bps|24bps|drc"] [dolby="surround"]
+                [samplerate="48khz|96khz"] [lang="\fIlanguage\fR"]
+                [content="normal|impaired|comments1|comments2"] />
+         [<audio ... />]
+         <subpicture [lang="\fIlanguage-code\fR"]>
+            <stream mode="normal|widescreen|letterbox|panscan"
+                [content="normal|large|children|normal_cc|large_cc|children_cc|forced|director|large_director|children_director"]
+                id="\fIstreamid\fR" />
+            [<stream ... />]
+         </subpicture>
+         [<subpicture ... />]
+         <pgc [entry="title"] [palette="\fIyuvfile|rgbfile\fR"]
+              [pause="\fIseconds\fR|inf"]>
+            <subpicture>
+               [<stream ... />]
+            </subpicture>
+            <pre> \fIcommands;\fR </pre>
+            <vob file="\fIfile.mpg\fR" [chapters="\fIchapter-list\fR"]
+                 [pause="\fIseconds\fR|inf"]>
+                 <cell [start="\fItimestamp\fR"] [end="\fItimestamp\fR"]
+                    [chapter="1|on|yes" | program="1|on|yes"] [pause="\fIseconds\fR|inf"]>
+                    \fIcommands;\fR
+                 </cell>
+            </vob>
+            [<vob ... />]
+            <button [name="\fIbuttonname\fR"]> \fIcommands;\fR </button>
+            [<button ... />]
+            <post> \fIcommands;\fR </post>
+         </pgc>
+         [<pgc ... />]
+      </menus>
+      [<menus ... />]
+   </vmgm>
+   <titleset>
+      <menus [lang="\fIlanguage-code\fR"]>
+         [<video ... />]
+         [<audio ... />]
+         <pgc [entry="\fIentries\fR"]
+              [palette="\fIyuvfile|rgbfile\fR"] [pause="\fIseconds\fR|inf"]>
+            [...]
+         </pgc>
+         [<pgc ... />]
+      </menus>
+      [<menus ... />]
+      <titles>
+         [<video ... />]
+         [<audio ... />]
+         <pgc [entry="notitle"] [palette="\fIyuvfile|rgbfile\fR"] [pause="\fIseconds\fR|inf"]>
+            [...]
+         </pgc>
+         [<pgc ... />]
+      </titles>
+   </titleset>
+   [<titleset ... />]
+</dvdauthor>
+.fi
+.PP
+A breakdown of the control file:
+.TP
+\fB<dvdauthor [dest="\fIoutput-dir\fB"] [jumppad="1|on|yes" | allgprm="1|on|yes"] [format="ntsc|pal"] [provider="\fIprovider\fB"]>\fR
+Initiates dvdauthor.  dest denotes the directory where \fBdvdauthor\fR will write the files. This is overridden by the -o option. Contains up to one <vmgm> tag and any number of <titleset>\&'s.
+.TP
+\fB<vmgm>\fR
+.TP
+\fB<titleset>\fR
+Constructs of a VMGM level menu set or a title set.
+Contains zero or more <menus> tags and if a
+titleset, up to one <titles> tag.
+.TP
+\fB<menus [lang="\fIlanguage-code\fB"]>\fR
+Marks a list of menus with a common language for this VMGM menu
+set or titleset, called in \fBdvdauthor\fR terminology a
+"pgcgroup."  Contains up to one <video> tag,
+up to one <audio> tag, up to one
+<subpicture> tag, and any number of
+<pgc> tags.
+.TP
+\fB<titles>\fR
+Marks the list of titles for this  titleset, called in
+\fBdvdauthor\fR terminology a "pgcgroup."  Contains up to
+one <video> tag, up to eight
+<audio> tags, up to 32
+<subpicture> tags, and any number of
+<pgc> tags.
+.TP
+\fB<video [format="ntsc|pal"] [aspect="4:3|16:9"] [resolution="\fIXxY\fB"] [caption="field1|field2"] [widescreen="nopanscan|noletterbox|crop"] />\fR
+Manually configures the video
+parameters for this pgcgroup.  If any of these are not set, then they
+will be inferred from the source stream.  Note that the DVD format
+only specifically supports 720x480, 704x480, 352x480, and 352x240
+resolutions for NTSC, and 720x576, 704x576, 352x576, and 352x288
+resolutions for PAL, but DVD author will accept a wider range of
+inputs and round up to the nearest size.
+.TP
+\fB<audio [format="mp2|ac3|dts|pcm"] [channels="\fInumchannels\fB"] [dolby="surround"] [quant="16bps|20bps|24bps|drc"] [samplerate="48khz|96khz"] [lang="\fIlanguage\fB"] [content="normal|impaired|comments1|comments2"] />\fR
+Manually configures an audio stream for this
+pgcgroup.  List once for each stream.  Most parameters are inferred
+automatically from the source VOBs except for PCM parameters.
+However, language and content must be manually specified.  Note that it is
+possible to just list the language and content attributes and let
+\fBdvdauthor\fR fill in the rest.
+.TP
+\fB<subpicture [lang="\fIlanguage\fB"] [content="normal|large|children|normal_cc|large_cc|children_cc|forced|director|large_director|children_director"] />\fR
+Manually configures a subpicture/subtitle for this pgcgroup or PGC.
+At the pgcgroup level, list once for each language. Occurrences at the PGC level
+don't have lang or content attributes; they inherit those
+from the corresponding <subpicture> tag at the pgcgroup level.
+.TP
+\fB<stream mode="normal|widescreen|letterbox|panscan" id="\fIstreamid\fB" />\fR
+Specifies the ID of a stream that is the representation of this
+subpicture in a particular display mode. This can be specified per-PGC, or
+pgcgroup-wide.
+.TP
+\fB<pgc [entry="\fIentries\fB"] [palette="\fIyuvfile|rgbfile\fB"] [pause="\fIseconds\fB|inf"]>\fR
+A PGC is just a fancy term for either a menu or a
+title.  It has a special meaning in the DVD spec so I have retained
+its use here.  PGC's can have commands that get executed before they
+start playing or after they finish; see <pre> and
+<post> tags below.
+
+If the PGC is a menu, you can specify one or more entries for
+it.  This means that if you press the corresponding button on your DVD
+remote, then it will go to this menu.  For a VMGM level menu, the only
+choice is title, which on my remote corresponds to
+the top menu button.  For a titleset level menu,
+you can use root, subtitle, audio, angle, and ptt.  If you want more
+than one, separate them by a space or a comma.  Note that
+root entry is meant for commands that jump from a
+VMGM level menu to a titleset menu.
+
+If the PGC is in a titleset, then it is assumed to be a title unless
+entry="notitle" is specified.
+
+All button and menu masks and all subtitles within a PGC must
+share the same 16 color palette.  If you use \fBspumux\fR
+to generate the subtitle/subpicture packets, then the color
+information will be automatically passed to
+\fBdvdauthor\fR; however, if you use another subtitler or
+want to have more control over the palette, you can manually specify
+it with the palette attribute.  The first 16
+entries of the file should be the 16 colors of the palette, listed as
+6 digit hexadecimal numbers representing either the RGB breakdown (if
+the filename ends in \&.rgb or the YUV breakdown (if
+the filename does not end in \&.rgb\&.  After that, the
+button group information can be listed as pairs of 8 digit hexadecimal
+numbers; up to three button groups may be specified.
+
+If you have a short video sequence or just want the video to
+pause at the end, you can use the pause attribute
+to set the number of seconds (as an integer) from 1 to 254.  If you
+want the video to pause indefinitely, use
+inf\&.
+.TP
+\fB<pre> \fIcommands;\fB </pre>\fR
+.TP
+\fB<post> \fIcommands;\fB </post>\fR
+Sets the commands to execute before or after a PGC
+plays.  It can be used to loop the current video (by having a
+<post> jump ... </post> sequence), or to
+conditionally skip certain chapters if a flag has been set.
+.TP
+\fB<fpc> \fIcommands;\fB </fpc>\fR
+Sets the commands to execute when the disk is first
+put in the player (FPC = First Program Chain).  It can be used to jump
+to a particular menu or initialize registers on startup.  If not
+specified, an implicit one will be created that jumps to the first
+menu found, or if there is no menu it will jump to the first title..
+.TP
+\fB<vob file="\fIfile.mpg\fB" [chapters="\fIchapter-list\fB"] [pause="\fIseconds\fB|inf"] />\fR
+Specifies an input video file for a menu or title, with optional chapter points and pause at the end.
+.TP
+\fB<cell [start="\fItimestamp\fB"] [end="\fItimestamp\fB"] [chapter="1|on|yes" | program="1|on|yes"] [pause="\fIseconds\fB|inf"]> \fIcommands;\fB </cell>\fR
+A more detailed way of specifying marker points in a title. If present, then the containing
+<vob> must not have a chapters attribute.
+A cell can have a VM command attached to it, to be executed when it plays. If the
+program attribute is set, then this cell will be a point that
+the user can skip to using the prev/next buttons on their DVD player remote. If the
+chapter attribute is set (implies program is set
+as well), then this cell is also a chapter point.
+.TP
+\fB<button [name="\fIbuttonname\fB"]> \fIcommands;\fB </button>\fR
+Specifies the commands to be executed when the user selects the button with the specified name.
+You define button names and placements with \fBspumux\fR\&.
+.SH "LANGUAGE DESCRIPTION"
+.PP
+The language is quite simple and roughly looks like C.
+.TP 0.2i
+\(bu
+Statements are terminated with a semicolon.
+.TP 0.2i
+\(bu
+Statements can span multiple lines.
+.TP 0.2i
+\(bu
+Multiple statements can appear on one line.
+.TP 0.2i
+\(bu
+Whitespace (space, tab, newlines) are not important, except to separate keywords and identifiers.
+.TP 0.2i
+\(bu
+C-style /* ... */ comments are allowed. Or you can use XML comments
+<!-- like this -->
+.SS "VARIABLES"
+.PP
+The DVD virtual machine processes 16 bit values.  It supports up
+to 16 general purpose registers; however \fBdvdauthor\fR
+reserves 3 for internal use.  Thus register 0-12 are available for
+use and are referred to as g0 through
+g12\&.
+.PP
+There are also 24 system registers, which can be referred to as
+s0 through s23\&.  Not all of
+these can be set.  Many of these have mnemonic synonyms.
+.TP
+\fBaudio (s1, rw)\fR
+Denotes the
+audio stream, ranging from 0-7.
+.TP
+\fBsubtitle (s2, rw)\fR
+The subtitle
+track, ranging from 0-31.  If you want the subtitle to always be
+displayed, then you should add 64 (i.e. choose 64-95).  Simply
+selecting the track (0-31) means that only the forced subtitles will
+be displayed, whereas enabling the track (64-95) means that all the
+subtitles will be displayed.  This allows you to have forced subtitles
+only for the parts of the movie where the actors are speaking a
+foreign (to the viewer) language, but still have normal subtitles for
+the hearing impaired.  The hearing impaired viewers would enable the
+track (64-95) while the other viewers would just select the track
+(0-31) they would be able to share the
+track.
+.TP
+\fBangle (s3, rw)\fR
+Selects the
+angle (currently untested).
+.TP
+\fBbutton (s8, rw)\fR
+Denotes the
+currently highlighted button.  Note that the value is multiplied by
+1024, so the first button is 1024, the second is 2048,
+etc.
+.SS "EXPRESSIONS"
+.PP
+Expressions follow typical C syntax except that booleans are not
+convertible to integers and vice versa.  Operators and comparisons
+are:
+.PP
+==, !=, >=, >, <=, <, &&, ||, !, eq, ne, ge, gt, le, lt, and, or, xor, not, +, -, *, /, %, &, |, ^
+.PP
+Since the code is encapsulated in XML, the parser will catch any
+unescaped < characters (i.e. not written as "&lt;"), thus alphabetic mnemonics have been
+provided for all comparison operators for consistency. Or you can put the code
+in a <![CDATA[ ... ]]> section.
+.PP
+There is also a numerical function:
+.TP
+\fBrandom(\fIEXPRESSION\fB)\fR
+Computes a pseudo-random number, between 1 and the
+supplied number, inclusively.
+.SS "BLOCKS"
+.PP
+Blocks are either a single statement (terminated by a
+semicolon), or a group of statements wrapped in curly braces.  For
+example:
+.TP 0.2i
+\(bu
+.nf
+g3 = s7;
+.fi
+.TP 0.2i
+\(bu
+.nf
+{
+  audio = 1;
+  subtitle = 65;
+  jump vmgm menu 3;
+}
+.fi
+.SS "STATEMENTS"
+.PP
+The statements supported are fairly simple at the moment.
+.TP
+\fB\fIVARIABLE\fB = \fIEXPRESSION\fB;\fR
+Sets a variable equal to the result of an equation.
+.TP
+\fBif (\fIEXPRESSION\fB) \fIBLOCK;\fB\fR
+.TP
+\fBif (\fIEXPRESSION\fB) \fIBLOCK;\fB else \fIBLOCK;\fB\fR
+Calculates the expression; if true, then it executes
+the block of code.
+.TP
+\fBjump \fITARGET\fB;\fR
+.TP
+\fBcall \fITARGET\fB [resume \fICELL\fB];\fR
+.TP
+\fBresume;\fR
+Jumps to a particular title or menu, or calls a
+particular menu, or returns to the calling title.  You can only
+execute a call from a title to a menu; all other forms are illegal.
+The purpose of using call instead of
+jump (besides the fact that they support a mutually
+exclusive list of targets) is to allow the menu to return to the point
+in the title where the call originated using
+resume\&.  You can manually specify the return cell
+by using the resume keyword, however if you do not
+specify one and you use the command in a post instruction block, then
+it will presume cell 1.
+.SS "TARGETS"
+.PP
+The following are possible targets (note that menus do not have
+chapters):
+.TP
+\fB[vmgm | titleset \fIX\fB] menu\fR
+.TP
+\fB[vmgm | titleset \fIX\fB] menu \fIY\fB\fR
+.TP
+\fB[vmgm | titleset \fIX\fB] menu entry \fIZ\fB\fR
+Targets either the default menu, a menu number Y, or the menu denoted as the entry for Z.  The menu is in either the VMGM or titleset domain.  If you wish to target a menu in the current domain then you can omit the domain moniker.
+.TP
+\fB[titleset \fIX\fB] title \fIY\fB [chapter \fIZ\fB]\fR
+Targets a title, or a chapter in a title.  Numbering
+starts at 1.  All of the titles on the disc are accessible in the VMGM
+domain, or you can access them by titleset
+instead.
+.TP
+\fBpgc \fIZ\fB\fR
+Targets a PGC in the current VMGM or titleset.
+.TP
+\fBchapter \fIZ\fB\fR
+Targets a chapter in the current title.
+.TP
+\fBprogram \fIZ\fB\fR
+.TP
+\fBcell \fIZ\fB\fR
+Targets a program or cell in the current PGC.  You can use
+this to create looping menus: jump cell
+1;
+.TP
+\fBcell top\fR
+.TP
+\fBnext cell\fR
+.TP
+\fBprev cell\fR
+.TP
+\fBprogram top\fR
+.TP
+\fBnext program\fR
+.TP
+\fBprev program\fR
+.TP
+\fBpgc top\fR
+.TP
+\fBnext pgc\fR
+.TP
+\fBprev pgc\fR
+.TP
+\fBup pgc\fR
+.TP
+\fBpgc tail\fR
+(Jump only) performs relative transfers of control within the current menu/title.
+"cell/program/pgc top" goes back to the start of the current cell/program/PGC;
+"next/prev cell/program/pgc" goes to the next or previous cell/program/PGC;
+"up pgc" goes to the "up" PGC (not currently settable in
+\fBdvdauthor\fR); and "pgc tail" goes to the
+<post> sequence in the current PGC.
+.PP
+Chapters are numbered from 1 in each title, while programs are numbered from
+1 in each PGC. Thus, the latter can reset independently of the former when there is more
+than one PGC in a title.
+.SH "LIMITATIONS"
+.PP
+The following limits are imposed by the DVD-Video spec.
+.PP
+There can be no more than 99 titlesets, no more than 99 menus in the VMG
+or a titleset, and no more than 99 titles in a titleset.
+.PP
+Each title may be made up of up to 999 PGCs. Each PGC may consist of up to 255 programs.
+The <pre> and <post> sections of a PGC
+put together can contain no more than 128 VM instructions.
+.PP
+Since there is only one VOB file (VIDEO_TS.VOB) in the VMG,
+the total amount of video in the VMG menus must fit into 1073709056 bytes (524272
+sectors of 2kiB each). In each titleset, all the menu video must fit in the first VOB
+(VTS_nn_0.VOB), so is limited to the same amount.
+<marc.leeman@gmail.com>
+MarcLeeman2003Marc Leeman
\ No newline at end of file
--- a/doc/dvddirdel.1
+++ b/doc/dvddirdel.1
@@ -0,0 +1,34 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "DVDDIRDEL" "1" "Fri Dec 30 19:47:26 CET 2005" "" "DVDAuthor Man Pages"
+
+.SH NAME
+dvddirdel \- Deletes a previously authored DVD directory structure in DIR
+.SH SYNOPSIS
+
+\fBdvddirdel\fR [ \fB-o \fIDIR\fB\fR ]
+
+.SH "DESCRIPTION"
+.PP
+\fBdvddirdel\fR safely deletes a previously authored DVD directory structure
+in DIR. Only items that look like part of a DVD-Video directory structure are removed.
+.PP
+If you just want to replace an existing output directory with a new one, dvdauthor(1)
+can do that directly with its -O option.
+.SH "USAGE"
+.PP
+\fBdvddirdel\fR -o DIR
+.SH "CAVEATS"
+.PP
+None.
+.SH "BUGS"
+.PP
+None.
+.SH "AUTHOR"
+.PP
+MarcLeemanThis manual page was written by Marc Leeman <marc.leeman@gmail.com>, for the Debian GNU/Linux system (but may be used by others).
+<marc.leeman@gmail.com>
+MarcLeeman2003Marc Leeman
\ No newline at end of file
--- a/doc/dvdunauthor.1
+++ b/doc/dvdunauthor.1
@@ -0,0 +1,35 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "DVDUNAUTHOR" "1" "Fri Dec 30 19:47:26 CET 2005" "" "DVDAuthor Man Pages"
+
+.SH NAME
+dvdunauthor \- Decodes DVD-Video file structure
+.SH SYNOPSIS
+
+\fBdvdunauthor\fR [ \fBpath\fR ]
+
+.SH "DESCRIPTION"
+.PP
+\fBdvdunauthor\fR decodes the specified DVD-Video directory structure,
+putting into the current directory an XML control file and corresponding MPEG-2
+files (with a .vob extension) that \fBdvdauthor\fR can (hopefully)
+interpret to recreate the DVD-Video directory structure.
+.SH "USAGE"
+.PP
+\fBdvdunauthor\fR path
+.SH "CAVEATS"
+.PP
+None.
+.SH "BUGS"
+.PP
+Depends on libdvdread to decode the DVD-Video structure and to report any
+errors trying to do so. But libdvdread will sometimes just exit without
+reporting any error.
+.SH "AUTHOR"
+.PP
+MarcLeemanThis manual page was written by Marc Leeman <marc.leeman@gmail.com>, for the Debian GNU/Linux system (but may be used by others).
+<marc.leeman@gmail.com>
+MarcLeeman2003Marc Leeman
\ No newline at end of file
--- a/doc/mpeg2desc.1
+++ b/doc/mpeg2desc.1
@@ -0,0 +1,50 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "MPEG2DESC" "1" "Fri Dec 30 19:47:26 CET 2005" "" "DVDAuthor Man Pages"
+
+.SH NAME
+mpeg2desc \- multiplex audio/video streams
+.SH SYNOPSIS
+
+\fBmpeg2desc\fR [ \fB-a \fIstream\fB\fR ] [ \fB-v \fIstream\fB\fR ] [ \fB-o \fIFILE\fB\fR ] [ \fB-s \fR ] [ \fB-m \fR ] [ \fB-h \fR ] [ \fB\fI< movie.mpg\fB\fR\fI ...\fR ]
+
+.SH "DESCRIPTION"
+.PP
+\fBmpeg2desc\fR multiplexes audio and video streams
+.SH "OPTIONS"
+.TP
+\fB-a \fIstream\fB\fR
+Output audio \fIstream\fR (0 .. 31) to stdout
+.TP
+\fB-v \fIstream\fB\fR
+Output video \fIstream\fR (0 .. 15) to stdout
+.TP
+\fB-o \fIFILE\fB\fR
+output the previous stream to \fIFILE\fR instead of stdout
+.TP
+\fB-s\fR
+skip to first valid header -- ensures mplex can handle output
+.TP
+\fB-m\fR
+output mplex offset to stdout
+.TP
+\fB-u\fR
+do not output messages about unrecognized headers
+.TP
+\fB-h\fR
+show short help
+.SH "USAGE"
+.PP
+\fBmpeg2desc\fR mpeg2desc [options] < movie.mpg
+.SH "CAVEATS"
+.PP
+None.
+.SH "BUGS"
+.PP
+None.
+.SH "AUTHOR"
+.PP
+MarcLeemanThis manual page was written by Marc Leeman <marc.leeman@gmail.com>, for the Debian GNU/Linux system (but may be used by others).
--- a/doc/spumux.1
+++ b/doc/spumux.1
@@ -0,0 +1,239 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "SPUMUX" "1" "Fri Dec 30 19:47:26 CET 2005" "" "DVDAuthor Man Pages"
+
+.SH NAME
+spumux \- generates and multiplexes subtitles into an existing mpeg2 program stream
+.SH SYNOPSIS
+
+\fBspumux\fR [ \fB-m dvd\fR | \fB-m cvd\fR | \fB-m svcd\fR ] [ \fB-s \fIstream\fB\fR ] [ \fB-v \fIlevel\fB\fR ] [ \fB-P\fR ] [ \fB--nomux\fR ] [ \fB--nodvdauthor-data\fR ] \fB\fIfile\fB\fR \fB< \fImpeg\fB\fR \fB> \fImpeg-with-subtitles\fB\fR
+
+.SH "DESCRIPTION"
+.PP
+\fBspumux\fR encodes the subtitles and multiplexes it into the mpeg2 program stream.
+.TP
+\fB-m \fImode\fB\fR
+Sets the encoding for the subtitles.  Can be dvd, cvd, or svcd.
+Default is dvd.
+.TP
+\fB-s \fIstream\fB\fR
+Sets the subtitle stream id.  Default is 0.
+.TP
+\fB-v \fIlevel\fB\fR
+Sets the verbosity level.
+.TP
+\fB-P\fR
+Enables a progress bar.
+.TP
+\fB--nomux\fR
+Disables reading of an MPEG stream from standard input. Instead, the output will
+contain only the subpicture stream.
+.TP
+\fB--nodvdauthor-data\fR
+Disables passing of color and button info to dvdauthor.
+.PP
+Here's a sample configuration file:
+
+.nf
+<subpictures>
+   <stream>
+      <spu start="\fIstart-time\fR" [ end="\fIend-time\fR" ] [ image="\fIpicture.png\fR" ]
+           [ highlight="\fIpicture.png\fR" ] [ select="\fIpicture.png\fR" ]
+           [ transparent="\fIcolor-spec\fR" ] [ force="yes" ]
+           [ autooutline="infer" [ outlinewidth="\fIwidth\fR" ]
+             [ autoorder="rows" | autoorder="columns" ] ]
+           [ xoffset="\fIx-coord\fR" yoffset="\fIy-coord\fR" ] >
+         <button [ name="\fIname\fR" ] [ x0="\fIx0\fR" ] [ y0="\fIy0\fR" ] [ x1="\fIx1\fR" ]
+                 [ y1="\fIy1\fR" ] [ up="\fIname\fR" ] [ down="\fIname\fR" ]
+                 [ left="\fIname\fR" ] [ right="\fIname\fR" ] />
+         <action [ name="\fIname\fR" ] [ x0="\fIx0\fR" ] [ y0="\fIy0\fR" ] [ x1="\fIx1\fR" ]
+                 [ y1="\fIy1\fR" ] />
+      </spu>
+   </stream>
+</subpictures>
+.fi
+.PP
+Each subtitle image is described by an <spu> tag.  The start and
+optional end attributes describe when the subtitle will be displayed
+on the screen, in "HR:MM:SS.HU".  The image attribute describes the
+main subtitle image, which can either be a PNG or BMP file.  If your
+image is not a full screen image, you can use xoffset and yoffset to
+move the picture around.  In the advent that the author is unable to
+use a graphics format with an alpha channel, then the transparent
+attribute can be used to describe which color should become fully
+transparent.  For the valid formats for \fIcolor-spec\fR,
+see SPECIFYING COLORS below.  If you want to
+force the display of the subtitle, regardless of whether the user has
+enabled subtitles or not, you can use the force tag.  When you are
+making menus, the force tag is required.
+.PP
+The remaining attributes and tags are related to menu creation.  The
+highlight attribute shows what all the buttons look like when they are
+highlighted (i.e. when you are using the arrows in the menu), and the
+select attribute shows what all the buttons look like when the are
+selected (i.e. for the 1-2 seconds after you press enter in the menu).
+If either of these (or the image attribute) are omitted, then spumux
+creates a blank (totally transparent) image.  Obviously at least one
+tag should be specified.
+.PP
+To aid in button creation, the autooutline attribute instructs spumux
+to infer where the buttons are located.  It does this by attempting to
+draw rectangles around a composition of the highlight and select
+images which do not intersect any opaque or semi-opaque pixels.  In
+order to support textual buttons, the attribute outlinewidth allows
+you to specify the width of the rectangle which is tested.  Wider
+rectangles won't be able to squeeze between the letters.  Finally,
+autoorder describes which way to order the automatically detected
+buttons, which is important for numerically selected buttons and for
+mapping buttons to button names or to the implied names.
+.PP
+The button and action tags describe the buttons (visibly selectable
+objects on the screen) and actions (commands that are executed as soon
+as the associated key is pressed on the remote).  If you are using the
+autooutline feature, just designing buttons, are happy with the
+inferred button navigation, and can deal with the simplistic naming
+system, then you actually do not even need to specify any buttons or
+actions.  Otherwise, read on.
+.PP
+The name attribute is used to give a button or action an easy to refer
+to name.  By default they are numbered sequentially starting with "1".
+The up, down, left, and right names describe which button or action
+should be tied to the corresponding key when the current button is
+highlighted, though if omitted spumux will use a reasonably
+intelligent algorithm to determine which buttons to move to.
+.PP
+The (x0,y0) coordinates describe the upper left hand corner
+inclusively, while the (x1,y1) coordinates describe the lower right
+hand corner EXclusively.  The coordinates start at 0,0 for the upper
+left hand corner.  Ideally, the y0 and y1 coordinates should both be
+even, so the button edges fully empasses two interlaced scanlines,
+even if there is no data in the extra scanline.  Some DVD players will
+fill in the extra scanline if it is not specified; spumux makes an
+effort to ensure that will be transparent but there is no guarantee.
+.SH "HANDLING TEXT-BASED SUBTITLES"
+.PP
+Spumux is also able to handle text subtitles, which will be rendered to
+graphics by spumux. A lot of different text (must be non-graphic) formats
+are supported (.sub, .srt, .ssa, .smi, .rt , .txt, .aqt, .jss, .js, ass).
+Spumux will try to determine the format automatically.
+.PP
+If processing text-based subtitles no other streams can be defined, buttons
+or others need to be processed in another pass with spumux, using another
+xml file.
+.PP
+Following .xml file shows the available tags and their default settings.
+Only the textsub tag is mandatory, defaults are used if the specific tag
+is not specified.
+
+.nf
+<subpictures [format="NTSC|PAL"]>
+   <stream>
+      <textsub filename="demo1.srt" characterset="UTF-8"
+         fontsize="28.0" font="arial" fill-color="\fIcolor-spec\fR"
+         outline-color="\fIcolor-spec\fR" outline-thickness="3.0"
+         shadow-offset="0, 0" shadow-color="\fIcolor-spec\fR"
+         horizontal-alignment="left"  vertical-alignment="bottom"
+         left-margin="60" right-margin="60"
+         top-margin="20" bottom-margin="30" subtitle-fps="\fIfps\fR"
+         movie-fps="\fIfps\fR"
+         movie-width="720" movie-height="\fIheight\fR" aspect="4:3"
+         force="yes"
+      />
+   </stream>
+</subpictures>
+.fi
+.PP
+The textsub tag defines the settings for the text to graphics rendering.
+.PP
+The filename attribute defines the path and name of the input text subtitle
+file, this is the only attribute that is mandatory.
+.PP
+The characterset attribute defines the characterset to be used, available
+charactersets can be found at http://www.gnu.org/software/libiconv.
+.PP
+The fontsize attribute defines the size of the font in font units.
+The font attribute defines the font used. Spumux will match the font name
+using Fontconfig if available, else it will use it as a filename on linux
+systems and will look into the windows/fonts and subdirectory spumux of
+the cygwin/mingw home directory and into the execution directory for windows-based
+execution for the specified font. Looking for free fonts? One listing is
+here http://www.microsoft.com/typography/links/links.asp?type=free&part=1
+(this is only about fonts, not about operating systems).
+.PP
+The fill-color attribute specifies the color to be used for filling the interior
+of the text, while outline-color specifies the color for the outline border to draw
+around each character, to set it off from the video background.
+The outline-thickness attribute specifies the thickness of this outline. For the
+valid formats for \fIcolor-spec\fR, see SPECIFYING COLORS below.
+The default fill color is white and the default outline color is black.
+.PP
+The shadow-offset and shadow-color attributes specify a shadow effect to be applied
+to the text. The shadow-offset is specified as 2 signed integers, being the horizontal and
+vertical offset respectively, with positive values being to the right and down. For the
+valid formats for \fIcolor-spec\fR, see SPECIFYING COLORS below.
+The default shadow color is black.
+.PP
+The horizontal-alignment attribute defines the horizontal alignment of the
+subtitles. Options are: "left", "right", "center" and "default".
+The "default" value causes spumux to use the attribute that is in the text
+subtitle file if the format supports such an attribute.
+.PP
+The vertical-alignment attribute defines the vertical alignment.
+Options are: "top", "center" and "bottom".
+.PP
+The margin attributes define the minimum blank pixel space between the
+border of the image and the border of the subtitle lines.
+.PP
+The subtitle-fps is needed for subtitle formats which use frame numbers rather
+than fractional seconds for specifying subtitle times, while the movie-fps
+is the frame rate of the movie onto which the subtitles are being multiplexed; if
+omitted, both of these default to 29.97 for NTSC and 25 for PAL.
+.PP
+Having separate subtitle-fps and movie-fps attributes make it possible for
+spumux to recalculate the subtitle timing if these are not the same.
+.PP
+The movie-width and movie-height attributes define the maximum size of
+the subtitle page, these shouldn't be larger than the size of the
+video frame (720x480 for NTSC, 720x576 for PAL), normally they are the same.
+Some DVD players prefer subtitle frames that are 2 or 4 pixels smaller in height.
+If omitted, the default movie width is 720 and the height is 2 pixels less than
+the video frame height.
+.PP
+The aspect attribute lets you specify whether the video aspect ratio is 16:9 (widescreen)
+or 4:3. This, together with the video format, is used to adjust the rendering of the
+text to ensure it appears undistorted.
+.PP
+The force option allows you to force the display of the subtitle,
+regardless of whether the user has enabled subtitles or not.
+.SH "SPECIFYING COLORS"
+.PP
+Colors for image transparency and text fill and outlines can be specified in a number
+of ways:
+.TP 0.2i
+\(bu
+As a color name, e.g. "green". You can use most of the color names
+listed at <http://www.imagemagick.org/script/color.php>\&.
+.TP 0.2i
+\(bu
+As a hexadecimal string preceded by a "#", similar to HTML format,
+e.g. "#FF8080". The number of digits must be a multiple of 3; the first group specifies
+the red component, the second green, and the third blue.
+.TP 0.2i
+\(bu
+As the name of a color space, followed by one to four component values
+in parentheses, e.g. "hsv(120, 50%, 90%)".
+.PP
+Valid color spaces are "gray" (1 component), "graya"
+(2 components), "hsl" (3 components), "hsla" (4 components), "hsv" (3 components),
+"hsva" (4 components), "rgb" (3 components) and "rgba" (4 components). Each component can
+be specified as either a percentage of the maximum value suffixed by "%", or an absolute
+value in the range 0-360 for hues, 0-255 for everything else. For color space names ending
+with "a", the last component is the alpha (transparency) value.
+.PP
+Note that alpha values are meaningless for the "transparent" attribute on the <spu>
+tag above.
+<marc.leeman@gmail.com>
+MarcLeeman2003Marc Leeman
\ No newline at end of file
--- a/doc/spuunmux.1
+++ b/doc/spuunmux.1
@@ -0,0 +1,60 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "SPUUNMUX" "1" "Fri Dec 30 19:47:26 CET 2005" "" "DVDAuthor Man Pages"
+
+.SH NAME
+spuunmux \- demultiplexes subtitles from an existing mpeg2 program stream
+.SH SYNOPSIS
+
+\fBspuunmux\fR [ \fB\fIinput file\fB\fR ] [ \fB\fIinput file\fB\fR\fI ...\fR ]
+
+.SH "DESCRIPTION"
+.PP
+\fBspuunmux\fR demultiplexes subtitles from an existing mpeg2 program stream
+.SH "OPTIONS"
+.TP
+\fB-o \fIname\fB\fR
+base name for script and images. Defaults to \fIsub\fR\&.
+.TP
+\fB-v \fIlevel\fB\fR
+verbosity level. \fIsub\fR\&. Defaults to \fI0\fR\&.
+.TP
+\fB-f\fR
+resize images to full size, which is 720x480 for NTSC, 720x576 for PAL.
+.TP
+\fB-F PAL|NTSC\fR
+specifies the video format, which is used to compute the dimensions of
+full size for -f.
+.TP
+\fB-s \fIstream\fB\fR
+number of the substream to extract. Defaults to \fI0\fR\&.
+.TP
+\fB-p \fIfile\fB\fR
+name of file with dvd palette. If palette file ends with .rgb, treated as a RGB else as a YCbCr color
+.TP
+\fB-h\fR
+print this help
+.TP
+\fB-V\fR
+print version number
+.SH "USAGE"
+.PP
+\fBspuunmux\fR inputfile1 inputfile2
+.PP
+demultiplexes subtitles from an existing mpeg2 program stream. To use stdin for demultiplexing, use - for the inputfile:
+.PP
+play_cell GoldFinger.iso 1 1 6 | \fBspuunmux\fR -
+.SH "CAVEATS"
+.PP
+None.
+.SH "BUGS"
+.PP
+None.
+.SH "AUTHOR"
+.PP
+MarcLeemanThis manual page was written by Marc Leeman <marc.leeman@gmail.com>, for the Debian GNU/Linux system (but may be used by others).
+<marc.leeman@gmail.com>
+MarcLeeman2016Marc Leeman
\ No newline at end of file
--- a/doc/video_format.7
+++ b/doc/video_format.7
@@ -0,0 +1,103 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "VIDEO_FORMAT" "7" "Fri Dec 30 19:47:26 CET 2005" "" "DVDAuthor Man Pages"
+
+.SH NAME
+video_format \- Proposal For A Video Format Preference Specification
+.SH "DESCRIPTION"
+.PP
+A number of different programs that create or work with video material need to know whether to generate their output in NTSC or PAL format. The ones that I’ve seen seem to have a hard-coded preference for one unless the user specifies the other. Overriding the default every time you want to use these programs gets tiresome.
+.PP
+Therefore, there needs to be a way for the user to configure a single default that will apply to all these programs. This default should be settable on a per-user basis and a systemwide basis, with the per-user setting overriding the systemwide one, as is usual. It makes sense to avoid dotfile clutter by following, at least to some extent, the XDG Base Directory Specification http://standards.freedesktop.org/basedir-spec/latest/\&.
+.PP
+The preference setting shall consist of the single word “NTSC” or “PAL” (without quotes, might as well be case-insensitive). This can be specified in the following ways, in order of decreasing priority:
+.TP 0.2i
+\(bu
+As the value of the environment variable VIDEO_FORMAT
+.TP 0.2i
+\(bu
+As the contents of the user-specific file $XDG_CONFIG_HOME/video_format, with $XDG_CONFIG_HOME defaulting to $HOME/.config if not defined.
+.TP 0.2i
+\(bu
+As the contents of a systemwide file \fIdir\fR/video_format, where \fIdir\fR is taken in turn from the colon-separated items in the value of $XDG_CONFIG_DIRS until the file is found, or if $XDG_CONFIG_DIRS is not defined, then the single directory /etc\&.
+.PP
+The application should use the first of these settings that it finds, searching in the specified order.
+.PP
+\fBNote:\fR the XDG Base Dir spec says systemwide config files should go in /etc/xdg\&. I'm not sure why this is, since lots of other apps put systemwide config in other places in /etc\&. Particularly since the spec doesn’t similarly restrict shared read-only data to just one subdirectory of /usr/share\&.
+.SH "DISCUSSION"
+.PP
+Can this spec apply to both standard-definition and high-definition video? For standard definition, there is a difference of frame size, being 720 by 480 for NTSC and 720 by 576 for PAL (in both cases, the aspect ratio of the displayed image is always 4:3). But this doesn’t seem to matter for high definition. For both standard-definition and high-definition, there is a difference of frame rates, based around 29.97 fps for NTSC and 25 fps for PAL.
+.PP
+Note also that if a config file earlier in the search contains an invalid setting, the search does not continue even if another, later config file might contain a valid setting. This allows the user to override a system default with “no default”. Is this important? I think it could be.
+.SH "REFERENCE IMPLEMENTATION"
+.PP
+The following Python code implements a routine called get_default_video_format that returns either one of the strings “NTSC” or “PAL” indicating the user/system-configured default format as per the spec above, or None if the default is invalid or not configured.
+
+.nf
+    class xdg_base_dir :
+        "Implementation of relevant parts of the XDG Base Directory specification" \\
+        " <http://standards.freedesktop.org/basedir-spec/latest/>."
+        @classmethod
+        def get_config_home(self) :
+            "returns the directory for holding user-specific config files."
+            result = os.environ.get("XDG_CONFIG_HOME")
+            if result == None :
+                result = os.path.join(os.environ["HOME"], ".config")
+            #end if
+            return result
+        #end get_config_home
+
+        @classmethod
+        def config_search_path(self) :
+            "returns the list of config directories to search (apart from the user area)."
+            return tuple(os.environ.get("XDG_CONFIG_DIRS", "/etc").split(":"))
+              # note spec actually says default should be /etc/xdg, but /etc is the
+              # conventional location for system config files.
+        #end config_search_path
+
+        @classmethod
+        def find_first_config_path(self, path) :
+            "searches for path in all the config directory locations in order of decreasing" \\
+            " priority, returning the expansion where it is first found, or None if not found."
+            paths_to_try = iter((self.get_config_home(),) + self.config_search_path())
+              # highest priority first
+            while True :
+                this_path = next(paths_to_try, None)
+                if this_path == None :
+                    break
+                this_path = os.path.join(this_path, path)
+                if os.path.exists(this_path) :
+                    break
+            #end while
+            return this_path
+        #end find_first_config_path
+
+    #end xdg_base_dir
+
+    valid_video_formats = frozenset(("NTSC", "PAL"))
+
+    def get_default_video_format() :
+        video_format = os.environ.get("VIDEO_FORMAT")
+        if video_format == None :
+            config_file = xdg_base_dir.find_first_config_path("video_format")
+            if config_file != None :
+                try :
+                    video_format = open(config_file, "r").readline().strip()
+                except OSError :
+                    video_format = None
+                #end try
+            #end if
+        #end if
+        if video_format != None :
+            video_format = video_format.upper()
+            if video_format not in valid_video_formats :
+                video_format = None
+            #end if
+        #end if
+        return video_format
+    #end get_default_video_format
+        
+.fi
